galpy/df/streamspraydf.py

```diff
@@ -124,6 +120,8 @@ def sample(self, n, return_orbit=True, returndt=False, integrate=True):
             If True, also return the time since the star was stripped. Default is False.
         integrate : bool, optional
             If True, integrate the orbits to the present time. If False, return positions at stripping (probably want to combine with returndt=True then to make sense of them!). Default is True.
+        pot_prog : galpy.potential.Potential or list of such instances or None, optional
+            Potential for the progenitor. Ignored if None.
```

Comment:
Since the potential of the progenitor and the ```MovingOrbitPotential`` that it gives rise to doesn't change, the progenitor potential and its ``MovingOrbitPotential`` counterpart should just initialized when the object is instantiated. And then you can just add it to ``_pot`` I think (but not ``_rtpot`` or ``_centerpot``) so it gets automatically used in the sampling routine.

Reply:
I think moving `pot_prog` to `__init__()` is a good idea. But I would not combine it with `self._pot`. Otherwise, the user would have to integrate the progenitor's orbit first, and then create a `MovingOrbitPotential` object, and finally pass it to the `__init__()` function. This is too much extra work. I still keep it as a non-moving potential, and create `MovingOrbitPotential` object when initialized.

Reply:
> I think moving `pot_prog` to `__init__()` is a good idea. But I would not combine it with `self._pot`. Otherwise, the user would have to integrate the progenitor's orbit first, and then create a `MovingOrbitPotential` object, and finally pass it to the `__init__()` function. This is too much extra work. I still keep it as a non-moving potential, and create `MovingOrbitPotential` object when initialized.

I didn't mean to put it in `pot` (I considered that, but decided against it for the reasons that you mention), but just that once you have integrated the progenitor in `pot` in the `__init__` function and you have set up the `MovingObjectPotential`, you can just add the `MovingObjectPotential` to `self._pot`, because it's only used to integrate stream stars after the initialization. That way you avoid having to check whether the progenitor should be included in the `sample` method.



> I noticed that my implementation of `pot_prog` doesn't work correctly if `center` or `centerpot` is set. In this case, I should replace `MovingObjectPotential(orbit=self._progenitor, ...)` with the true orbit considering the center. Is there a easy way to combine `self._progenitor` and `self._center` into a new `Orbit` object?

I believe the progenitor's orbit integrated in `pot` is the true orbit even when `center` is set (see the `test_center` code). How does it not work?

Reply:
> I believe the progenitor's orbit integrated in pot is the true orbit even when center is set (see the test_center code). How does it not work?

I misunderstood this. Then it should work!